api/src/main/java/io/opentelemetry/trace/propagation/JaegerPropagator.java

```diff
@@ -0,0 +1,200 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.propagation;
+
+import static io.opentelemetry.internal.Utils.checkNotNull;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.internal.StringUtils;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Implementation of the Jaeger propagation protocol. See <a
+ * href=https://www.jaegertracing.io/docs/client-libraries/#propagation-format>Jaeger Propogation
+ * Format</a>.
+ */
+@Immutable
+public class JaegerPropagator implements HttpTextFormat {
+
+  private static final Logger logger = Logger.getLogger(HttpTraceContext.class.getName());
+
+  static final String TRACE_ID_HEADER = "uber-trace-id";
+  // Parent span has been deprecated but Jaeger propagation protocol requires it
+  static final String DEPRECATED_PARENT_SPAN = "0";
+  static final String SEPARATOR = ":";
+
+  private static final String IS_SAMPLED = "1";
+  private static final String NOT_SAMPLED = "0";
+
+  private static final int MAX_TRACE_ID_LENGTH = 2 * TraceId.getSize();
+  private static final int MAX_SPAN_ID_LENGTH = 2 * SpanId.getSize();
+  private static final int MAX_FLAGS_LENGTH = 2;
+  private static final TraceFlags SAMPLED_FLAGS = TraceFlags.builder().setIsSampled(true).build();
+  private static final TraceFlags NOT_SAMPLED_FLAGS =
+      TraceFlags.builder().setIsSampled(false).build();
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_ID_HEADER);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, C carrier, Setter<C> setter) {
+    checkNotNull(context, "context");
+    checkNotNull(setter, "setter");
+    checkNotNull(carrier, "carrier");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
+    if (span == null) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+    String sampled = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+
+    setter.set(
+        carrier,
+        TRACE_ID_HEADER,
+        spanContext.getTraceId().toLowerBase16()
+            + SEPARATOR
+            + spanContext.getSpanId().toLowerBase16()
+            + SEPARATOR
+            + DEPRECATED_PARENT_SPAN
+            + SEPARATOR
+            + sampled);
```

Comment:
consider to benchmark this to see if allocated a char[] with the right size and calling `copyLowerBase16To` gives better performance. maybe just a TODO.

Reply:
I had a look into creating jmh benchmarks for the import and export of the context using ```JaegerPropagator``` - I think they will look similar to the ones already written for ```HttpTraceContext```.
Worth raising a seperate issue to add these?

Reply:
@jpkrohling Yeap, it will be clearer if u add it as separate issue, so it will be easier to track